src/rpc/rawtransaction.cpp

```diff
@@ -1037,13 +1037,25 @@ UniValue signrawtransaction(const UniValue& params, bool fHelp)
         }
     }
 
+    std::vector<CTxOut> allPrevOutputs;
+    // We do not need to know the inputs for pre-v5 transactions.
+    // We can't sign v5+ transactions without knowing all inputs.
+    if (mergedTx.nVersion >= ZIP225_TX_VERSION) {
```

Comment:
This almost makes me want for there to be separate `PrecomputedTransactionData` types for pre- and post-V5. The meaning of the empty set of `allPrevOutputs` is now overloaded; it both means "we don't need this for `v4`", and "this set is required to be empty when signing coinbase transactions in V5".